ui-modules/utils/logbook/logbook.js

```diff
@@ -39,168 +39,249 @@ export function logbook() {
-                if ($scope.isLatest && $scope.logEntries.length !== 0) {
-                    if (logEntries.length > 0) {
+                // Assign unique IDs for new log entries.
+                logEntries.forEach(item => item.id = generateLogEntryId());
+
+                if (logEntries.length > 0 && isTail() && $scope.autoRefresh && !isNewQueryParameters) {
 
-                        // Calculate date-time to display up to. Note, calendar does not take into account milliseconds,
-                        // round down to seconds.
-                        let latestDateTimeToDisplay = Math.floor(logEntries.slice(-1)[0].datetime / DEFAULT_NUMBER_OF_ITEMS) * DEFAULT_NUMBER_OF_ITEMS;
+                    // Calculate date-time to display up to. Note, calendar does not take into account milliseconds, round down to seconds.
+                    let dateTimeOfLastLogEntry = Math.floor(logEntries.slice(-1)[0].datetime / DEFAULT_NUMBER_OF_ITEMS) * DEFAULT_NUMBER_OF_ITEMS;
+                    let dateTimeFrom = new Date($scope.search.dateTimeFrom).getTime();
+
+                    if (dateTimeOfLastLogEntry > dateTimeFrom) {
 
                         // Display new log entries.
-                        let newLogEntries = logEntries.filter(entry => entry.datetime <= latestDateTimeToDisplay);
+                        let newLogEntries = logEntries.filter(({datetime}) => datetime <= dateTimeOfLastLogEntry);
                         $scope.logEntries = $scope.logEntries.concat(newLogEntries).slice(-DEFAULT_NUMBER_OF_ITEMS);
 
-                        // Cache next date-time to query tail from.
-                        dateTimeToAutoUpdateFrom = new Date(latestDateTimeToDisplay);
+                        // Optimize auto-refresh: cache next date-time to query tail from, if it is still a tail.
+                        dateTimeToAutoRefreshFrom = dateTimeOfLastLogEntry;
+
                     } else {
-                        // Or re-set the cache.
-                        dateTimeToAutoUpdateFrom = '';
+                        // Or re-set the cached value.
+                        dateTimeToAutoRefreshFrom = '';
                     }
                 } else {
                     $scope.logEntries = logEntries;
                 }
 
-                $scope.logtext = covertLogEntriesToString($scope.logEntries);
-                scrollToMostRecentRecords();
+                // Auto-scroll.
+                if ($scope.logEntries.length > 0) {
+                    if ($scope.isAutoScrollDown) {
+                        scrollToMostRecentLogEntry();
+                    } else if (datetimeToScrollTo && datetimeToScrollTo >= $scope.logEntries[0].datetime) {
+                        scrollToLogEntryWithDateTime(datetimeToScrollTo);
+                    }
+                }
+
+                // Re-set marker for search parameters changes after auto-scroll.
+                isNewQueryParameters = false;
+
+                if ($scope.logEntries.length === 0) {
```

Comment:
could be an `else { ..}` to the previous if for brevity

Reply:
No, `isNewQueryParameters` must be reset before that, regardless of query result.